ci/Jenkinsfile.premerge

```diff
@@ -0,0 +1,286 @@
+#!/usr/local/env groovy
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ *
+ * Jenkinsfile for building spark-rapids-jni on blossom
+ *
+ */
+import hudson.model.Result
+import hudson.model.Run
+import jenkins.model.CauseOfInterruption.UserInterruption
+
+@Library(['blossom-lib']) _
+@Library('blossom-github-lib@master')
+import ipp.blossom.*
+
+def githubHelper // blossom github helper
+def TEMP_IMAGE_BUILD = true
+def IMAGE_PREMERGE = "${common.ARTIFACTORY_NAME}/sw-spark-docker/plugin-jni:centos7-cuda11.5.0-blossom"
+def cpuImage = pod.getCPUYAML(IMAGE_PREMERGE)
+def PREMERGE_DOCKERFILE = 'ci/Dockerfile'
+def PREMERGE_TAG
+def skipped = false
+def major_ver // major version extracted from project version
+def minor_ver // minor version extracted from project version
+
+pipeline {
+    agent {
+        kubernetes {
+            label "premerge-init-${BUILD_TAG}"
+            cloud 'sc-ipp-blossom-prod'
+            yaml cpuImage
+        }
+    }
+
+    options {
+        ansiColor('xterm')
+        buildDiscarder(logRotator(numToKeepStr: '50'))
+        skipDefaultCheckout true
+        timeout(time: 12, unit: 'HOURS')
+    }
+
+    parameters {
+        string(name: 'PARALLEL_LEVEL', defaultValue: '18',
+            description: 'Parallel build cudf cpp with -DCPP_PARALLEL_LEVEL')
+        string(name: 'REF', defaultValue: '',
```

Comment:
Did not found where this parameter is used?

Reply:
> Generally copied from internal pre-merge jenkinsfile, but modified params to support being triggered by upstream router job

Reply:
So it will be used in pipeline configure to fetch jenkinsfile based on input $REF

Reply:
Got it